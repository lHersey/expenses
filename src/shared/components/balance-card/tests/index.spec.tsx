import { render } from '@testing-library/react-native';
import React from 'react';
import Providers from 'shared/providers/test-providers';

import BalanceCard from '../index';

describe('<BalanceCard />', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
      expect.hasAssertions();
      const rendered = render(
        <Providers>
          <BalanceCard />
        </Providers>,
      ).toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });
});
