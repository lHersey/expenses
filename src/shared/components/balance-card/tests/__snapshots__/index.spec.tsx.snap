// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BalanceCard /> snapshots should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "elevation": 8,
          "overflow": "visible",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 4,
        },
        Object {
          "display": "flex",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4280987875,
            4292112360,
            4294874740,
          ],
          "endPoint": Array [
            1,
            0,
          ],
          "locations": undefined,
          "startPoint": Array [
            0,
            0,
          ],
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#FFF",
            "borderRadius": 15,
            "display": "flex",
            "flexGrow": 1,
            "overflow": "hidden",
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "marginTop": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#3e3e3e",
                "fontSize": 16,
              },
              Object {
                "color": "#e3e3e3",
              },
            ]
          }
        >
          Total Balance
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "marginTop": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#3e3e3e",
                "fontSize": 16,
              },
              Object {
                "color": "#e3e3e3",
                "fontSize": 36,
                "fontWeight": "600",
              },
            ]
          }
        >
          $ 4800.00
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginTop": 20,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.15)",
                "borderRadius": 10,
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 10,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 10,
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
            ]
          }
        >
          <Text />
          <View
            style={
              Array [
                Object {
                  "marginLeft": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#3e3e3e",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e3e3e3",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Income
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#3e3e3e",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e3e3e3",
                    "fontSize": 14,
                  },
                ]
              }
            >
              $
              4309
              .00
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.15)",
                "borderRadius": 10,
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 10,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 10,
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
            ]
          }
        >
          <Text />
          <View
            style={
              Array [
                Object {
                  "marginLeft": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#3e3e3e",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e3e3e3",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Expenses
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#3e3e3e",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e3e3e3",
                    "fontSize": 14,
                  },
                ]
              }
            >
              $
              400
              .00
            </Text>
          </View>
        </View>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</RNCSafeAreaProvider>
`;
